vm/class.go

```diff
@@ -60,12 +61,16 @@ type BaseClass struct {
 	Methods *environment
 	// ClassMethods contains this class's methods
 	ClassMethods *environment
-	// SuperClass points to the class it inherits
-	SuperClass *RClass
+	// pseudoSuperClass points to the class it inherits
+	pseudoSuperClass *RClass
+	// This is the class where we should looking for a method.
+	// It can be normal class, singleton class or a module.
+	superClass *RClass
```

Comment:
Why we have two instance `pseudoSuperClass` and `superClass`? In `Integer`, `String` etc this variables have the same object.